### ISSUE TITLE
Performance decay "ipopt" + "idas" in 3.6.0

### ISSUE THREAD

**Issue by kozachynskyi** [casadi/casadi#3121]
Hello everyone,

If you run the example below with casadi 3.5.5 and 3.6.0, you'll notice that ipopt takes more steps with idas in 3.6.0. 

In the example below, the objective function is Parameter Estimation Ordinary Least Squares, based only on state variables. So I don't think it has anything to do with issue #2806. There is no noticeable performance difference with cvodes. So I'm assuming that the ipopt version has no influence here, and it has nothing. @jaeandersson Let me know if I can help in any way.

```python
from casadi import *
from numpy import *
from pylab import *

# Example taken from IDAS Tutorial

# http://casadi.sourceforge.net/api/html/db/d58/integrators_2idas_8py-example.html
# for DAE
g = MX.sym("g")

def fun_eval_dae(fixed, integrator_name):
    L = 5.0

    #! differential states
    x = MX.sym("x")
    y = MX.sym("y")
    u = MX.sym("u")
    v = MX.sym("v")

    #! algebraic states
    lambd = MX.sym("lambda")

    #! All states and parameters
    x_all = vertcat(x, u, y, v)
    z_all = lambd

    #! We construct the DAE system
    ode = vertcat(u, lambd * x)
    alg = vertcat(
        x**2 + y**2 - L**2,
        u * x + v * y,
        u**2 - g * y + v**2 + L**2 * lambd,
    )
    x_all = vertcat(x, u)
    z_all = vertcat(y, v, lambd)
    dae = {"x": x_all, "z": z_all, "p": vertcat(g), "ode": ode, "alg": alg}

    #! the initial state of the pendulum
    X_ = [3, -1.0 / 3]
    Z_ = [4, 1.0 / 4, 1147.0 / 720]

    opts = {"t0": 0, "tf": 0.1}

    I = integrator("I", integrator_name, dae, opts)
    x0 = X_
    z0 = Z_
    all_x = []
    all_z = []

    if fixed:
        p = 2
    else:
        p = g

    for i in range(8):
        one_step = I(p=p, x0=x0, z0=z0)
        x0 = one_step["xf"]
        z0 = one_step["zf"]
        all_x.append(x0)
        all_z.append(z0)

    res = {"x": hcat(all_x), "z": hcat(all_z)}
    return res

u = MX.sym("u")

def fun_eval_ode(fixed, integrator_name):
    # State
    x = MX.sym("x",3)
    s = x[0] # position
    v = x[1] # speed
    m = x[2] # mass

    X_ = [0,0,1]

    # ODE right hand side
    sdot = v
    vdot = (u - 0.05 * v*v)/m
    mdot = -0.1*u*u
    xdot = vertcat(sdot,vdot,mdot)

    ode = {'x':x, 'p':vertcat(u), 'ode':xdot}
    opts = {"t0": 0, "tf": 0.1}

    I = integrator('I', "cvodes", ode, opts)
    x0=X_
    all_x = []

    if fixed:
        p = 2
    else:
        p = u

    for i in range(8):
        one_step = I(p = p, x0=x0)
        x0=one_step["xf"]
        all_x.append(x0)

    res = {"x": hcat(all_x)}
    return res

integrator_name = "idas"
for fun, variable in zip([fun_eval_ode, fun_eval_dae], [u, g]) :
    obj = fun(False, integrator_name)["x"] - fun(True, integrator_name)["x"]
    solver = nlpsol("s", "ipopt", dict(x=variable, f=sumsqr(obj)))
    print(solver(x0=[1], lbx=[1], ubx=[3]))

```

**Comment by jaeandersson** [comment]
Here is the output I get if I try to run it:
```
CasADi - 2023-04-19 05:47:23 WARNING("The options 't0', 'tf', 'grid' and 'output_t0' have been deprecated.
The same functionality is provided by providing additional input arguments to the 'integrator' function, in particular:
 * Call integrator(..., t0, tf, options) for a single output time, or
 * Call integrator(..., t0, grid, options) for multiple grid points.
The legacy 'output_t0' option can be emulated by including or excluding 't0' in 'grid'.
Backwards compatibility is provided in this release only.") [.../casadi/core/integrator.cpp:499]

******************************************************************************
This program contains Ipopt, a library for large-scale nonlinear optimization.
 Ipopt is released as open source code under the Eclipse Public License (EPL).
         For more information visit https://github.com/coin-or/Ipopt
******************************************************************************

This is Ipopt version 3.14.11, running with linear solver MUMPS 5.5.1.

Number of nonzeros in equality constraint Jacobian...:        0
Number of nonzeros in inequality constraint Jacobian.:        0
Number of nonzeros in Lagrangian Hessian.............:        1

Total number of variables............................:        1
                     variables with only lower bounds:        0
                variables with lower and upper bounds:        1
                     variables with only upper bounds:        0
Total number of equality constraints.................:        0
Total number of inequality constraints...............:        0
        inequality constraints with only lower bounds:        0
   inequality constraints with lower and upper bounds:        0
        inequality constraints with only upper bounds:        0

iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
   0  3.5792669e+00 0.00e+00 6.26e+00  -1.0 0.00e+00    -  0.00e+00 0.00e+00   0
   1  2.6614383e+00 0.00e+00 4.73e+00  -1.0 1.55e-01    -  1.53e-01 1.00e+00f  1
   2  7.7825285e-01 0.00e+00 4.92e+00  -1.0 1.19e+00    -  1.00e+00 1.00e+00f  1
   3  1.4286333e-01 0.00e+00 1.69e+00  -1.0 1.94e-01    -  9.43e-01 1.00e+00f  1
   4  1.1320624e-02 0.00e+00 5.10e-01  -1.0 1.13e-01    -  1.00e+00 1.00e+00f  1
   5  5.1476899e-04 0.00e+00 9.76e-02  -1.7 3.63e-02    -  1.00e+00 1.00e+00f  1
   6  1.6243907e-05 0.00e+00 1.81e-02  -2.5 8.17e-03    -  1.00e+00 1.00e+00f  1
   7  4.6712955e-07 0.00e+00 3.11e-03  -3.8 1.47e-03    -  1.00e+00 1.00e+00f  1
   8  1.3194884e-08 0.00e+00 5.23e-04  -3.8 2.50e-04    -  1.00e+00 1.00e+00f  1
   9  3.7165983e-10 0.00e+00 8.79e-05  -5.7 4.20e-05    -  1.00e+00 1.00e+00f  1
iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
  10  1.0435940e-11 0.00e+00 1.47e-05  -5.7 7.04e-06    -  1.00e+00 1.00e+00f  1
  11  2.8934879e-13 0.00e+00 2.46e-06  -8.6 1.18e-06    -  1.00e+00 1.00e+00f  1
  12  7.8306448e-15 0.00e+00 4.05e-07  -8.6 1.97e-07    -  1.00e+00 1.00e+00f  1
  13  2.1186971e-16 0.00e+00 6.66e-08  -8.6 3.25e-08    -  1.00e+00 1.00e+00f  1
  14  5.7324496e-18 0.00e+00 1.10e-08  -8.6 5.34e-09    -  1.00e+00 1.00e+00f  1
  15  1.5510112e-19 0.00e+00 1.80e-09  -9.0 8.78e-10    -  1.00e+00 1.00e+00f  1

Number of Iterations....: 15

                                   (scaled)                 (unscaled)
Objective...............:   1.5510112472728868e-19    1.5510112472728868e-19
Dual infeasibility......:   1.8020376749521422e-09    1.8020376749521422e-09
Constraint violation....:   0.0000000000000000e+00    0.0000000000000000e+00
Variable bound violation:   0.0000000000000000e+00    0.0000000000000000e+00
Complementarity.........:   9.0909091049303121e-10    9.0909091049303121e-10
Overall NLP error.......:   1.8020376749521422e-09    1.8020376749521422e-09


Number of objective function evaluations             = 16
Number of objective gradient evaluations             = 16
Number of equality constraint evaluations            = 0
Number of inequality constraint evaluations          = 0
Number of equality constraint Jacobian evaluations   = 0
Number of inequality constraint Jacobian evaluations = 0
Number of Lagrangian Hessian evaluations             = 15
Total seconds in IPOPT                               = 0.375

EXIT: Optimal Solution Found.
           s  :   t_proc      (avg)   t_wall      (avg)    n_eval
       nlp_f  |  87.25ms (  5.45ms)  17.91ms (  1.12ms)        16
  nlp_grad_f  | 385.83ms ( 22.70ms)  57.95ms (  3.41ms)        17
  nlp_hess_l  |   1.32 s ( 87.86ms) 275.68ms ( 18.38ms)        15
       total  |   2.25 s (  2.25 s) 378.83ms (378.83ms)         1
{'f': DM(1.55101e-19), 'g': DM(0x1), 'lam_g': DM(0x1), 'lam_p': DM(0x0), 'lam_x': DM(-2.06718e-17), 'x': DM(2)}
This is Ipopt version 3.14.11, running with linear solver MUMPS 5.5.1.

Number of nonzeros in equality constraint Jacobian...:        0
Number of nonzeros in inequality constraint Jacobian.:        0
Number of nonzeros in Lagrangian Hessian.............:        1

Total number of variables............................:        1
                     variables with only lower bounds:        0
                variables with lower and upper bounds:        1
                     variables with only upper bounds:        0
Total number of equality constraints.................:        0
Total number of inequality constraints...............:        0
        inequality constraints with only lower bounds:        0
   inequality constraints with lower and upper bounds:        0
        inequality constraints with only upper bounds:        0

iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
   0  5.1164534e-01 0.00e+00 1.04e+00  -1.0 0.00e+00    -  0.00e+00 0.00e+00   0
   1  4.0424465e-01 0.00e+00 3.92e-01  -1.0 1.09e-01    -  5.12e-01 1.00e+00f  1
   2  1.2983017e-03 0.00e+00 6.29e-01  -1.7 2.86e+00    -  1.20e-01 3.25e-01f  2
   3  7.4238097e-04 0.00e+00 5.12e-02  -1.7 8.83e-02    -  9.37e-01 1.00e+00f  1
   4  3.6648001e-04 0.00e+00 3.55e-02  -2.5 1.61e+00    -  3.97e-01 4.01e-02f  5
   5  1.4389935e-05 0.00e+00 6.17e-03  -2.5 3.35e+00    -  2.44e-01 9.55e-03f  6
   6  3.3755642e-06 0.00e+00 2.69e-03  -3.8 6.72e+00    -  1.32e-01 1.17e-03f  8
   7  9.4203617e-07 0.00e+00 1.39e-03  -3.8 5.91e+01    -  1.73e-02 6.62e-05f  9
   8  1.8657429e-07 0.00e+00 6.20e-04  -5.7 1.25e+02    -  7.99e-03 1.57e-05f 10
   9  7.2214894e-08 0.00e+00 3.86e-04  -5.7 6.43e+00  -4.0 1.00e+00 1.52e-04f 11
iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
  10  6.6469516e-09 0.00e+00 1.20e-04  -5.7 6.66e+01    -  1.50e-02 7.33e-06f 12
  11  3.6147992e-11 0.00e+00 8.68e-06  -5.7 4.54e+01    -  2.18e-02 2.69e-06f 14
  12  2.4297040e-11 0.00e+00 7.08e-06  -8.6 9.40e+01    -  1.05e-02 1.62e-07f 17
  13  1.9212998e-11 0.00e+00 6.30e-06  -8.6 2.13e-01  -4.5 1.00e+00 6.10e-05f 15
  14  1.1826252e-12 0.00e+00 1.58e-06  -8.6 1.48e+02    -  6.77e-03 5.16e-08f 18
  15  2.1109422e-13 0.00e+00 6.64e-07  -8.6 1.41e-01  -5.0 1.00e+00 1.53e-05f 17
  16  5.0350856e-14 0.00e+00 3.28e-07  -8.6 8.04e+01    -  1.24e-02 1.19e-08f 21
  17  1.3813266e-14 0.00e+00 1.70e-07  -8.6 1.93e+02    -  5.18e-03 2.47e-09f 22
  18  2.8547683e-15 0.00e+00 7.93e-08  -8.6 1.66e+02    -  1.00e+00 1.44e-09f 23
  19  1.0271963e-15 0.00e+00 4.70e-08  -8.6 3.22e+01    -  3.11e-02 3.70e-09f 24
iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
  20  1.1427751e-16 0.00e+00 1.55e-08  -8.6 4.44e+01    -  2.38e-02 1.34e-09f 25
  21  5.0297661e-19 0.00e+00 1.02e-09  -9.0 4.17e-03  -5.4 1.00e+00 3.81e-06f 19

Number of Iterations....: 21

                                   (scaled)                 (unscaled)
Objective...............:   5.0297660594016314e-19    5.0297660594016314e-19
Dual infeasibility......:   1.0242224752276239e-09    1.0242224752276239e-09
Constraint violation....:   0.0000000000000000e+00    0.0000000000000000e+00
Variable bound violation:   0.0000000000000000e+00    0.0000000000000000e+00
Complementarity.........:   9.0909089464684192e-10    9.0909089464684192e-10
Overall NLP error.......:   1.0242224752276239e-09    1.0242224752276239e-09


Number of objective function evaluations             = 357
Number of objective gradient evaluations             = 22
Number of equality constraint evaluations            = 0
Number of inequality constraint evaluations          = 0
Number of equality constraint Jacobian evaluations   = 0
Number of inequality constraint Jacobian evaluations = 0
Number of Lagrangian Hessian evaluations             = 21
Total seconds in IPOPT                               = 2.550

EXIT: Optimal Solution Found.
           s  :   t_proc      (avg)   t_wall      (avg)    n_eval
       nlp_f  |   1.00 s (  2.81ms)   1.00 s (  2.81ms)       357
  nlp_grad_f  | 249.73ms ( 10.86ms) 249.72ms ( 10.86ms)        23
  nlp_hess_l  |   1.28 s ( 60.76ms)   1.28 s ( 60.76ms)        21
       total  |   2.55 s (  2.55 s)   2.55 s (  2.55 s)         1
{'f': DM(5.02977e-19), 'g': DM(0x1), 'lam_g': DM(0x1), 'lam_p': DM(0x0), 'lam_x': DM(-2.48043e-13), 'x': DM(2)}

```

**Comment by kozachynskyi** [comment]
1. For IDAS (second output) I get the same result with 3.6.0 and 3.6.1. Can you try with 3.5.5?
2. CVODES now gives different output in 3.6.0 and 3.6.1. I've created a separate issue #3123  if you prefer to have everything in one issue -> let me know and I'll merge it.

**Comment by jaeandersson** [comment]
The `integrator_name` argument is ignored in `fun_eval_ode` and "cvodes" is hardcoded in the call to `casadi.integrator`. If I replace  it with 'idas' I get a segfault(!). I'll open a separate issue for that.

**Comment by jaeandersson** [comment]
@kozachynskyi If you're able to, you can try out the [issue_3130](https://github.com/casadi/casadi/compare/issue_3130) branch. It resolves the issue with incorrect evaluation in `jtimesF` and related segfault in IDAS.

**Comment by kozachynskyi** [comment]
> The `integrator_name` argument is ignored in `fun_eval_ode` and "cvodes" is hardcoded in the call to `casadi.integrator`. If I replace it with 'idas' I get a segfault(!). I'll open a separate issue for that.

Here is a modified example without hardcoded "cvodes". This code can be run in casadi 3.5.5 and with version without segmentation fault https://github.com/casadi/casadi/compare/issue_3130

```python
from casadi import *
from numpy import *
from pylab import *
import casadi as ca

# Example taken from IDAS Tutorial

# http://casadi.sourceforge.net/api/html/db/d58/integrators_2idas_8py-example.html
# for DAE
g = MX.sym("g")

def fun_eval_dae(fixed, integrator_name):
    L = 5.0

    #! differential states
    x = MX.sym("x")
    y = MX.sym("y")
    u = MX.sym("u")
    v = MX.sym("v")

    #! algebraic states
    lambd = MX.sym("lambda")

    #! All states and parameters
    x_all = vertcat(x, u, y, v)
    z_all = lambd

    #! We construct the DAE system
    ode = vertcat(u, lambd * x)
    alg = vertcat(
        x**2 + y**2 - L**2,
        u * x + v * y,
        u**2 - g * y + v**2 + L**2 * lambd,
    )
    x_all = vertcat(x, u)
    z_all = vertcat(y, v, lambd)
    dae = {"x": x_all, "z": z_all, "p": vertcat(g), "ode": ode, "alg": alg}

    #! the initial state of the pendulum
    X_ = [3, -1.0 / 3]
    Z_ = [4, 1.0 / 4, 1147.0 / 720]

    opts = {"t0": 0, "tf": 0.1}

    I = integrator("I", integrator_name, dae, opts)
    x0 = X_
    z0 = Z_
    all_x = []
    all_z = []

    if fixed:
        p = 2
    else:
        p = g

    for i in range(8):
        one_step = I(p=p, x0=x0, z0=z0)
        x0 = one_step["xf"]
        z0 = one_step["zf"]
        all_x.append(x0)
        all_z.append(z0)

    res = {"x": hcat(all_x), "z": hcat(all_z)}
    return res

u = MX.sym("u")

def fun_eval_ode(fixed, integrator_name):
    # State
    x = MX.sym("x",3)
    s = x[0] # position
    v = x[1] # speed
    m = x[2] # mass

    X_ = [0,0,1]

    # ODE right hand side
    sdot = v
    vdot = (u - 0.05 * v*v)/m
    mdot = -0.1*u*u
    xdot = vertcat(sdot,vdot,mdot)

    ode = {'x':x, 'p':vertcat(u), 'ode':xdot}
    opts = {"t0": 0, "tf": 0.1}

    I = integrator('I', integrator_name, ode, opts)
    x0=X_
    all_x = []

    if fixed:
        p = 2
    else:
        p = u

    for i in range(8):
        one_step = I(p = p, x0=x0)
        x0=one_step["xf"]
        all_x.append(x0)

    res = {"x": hcat(all_x)}
    return res

integrator_name = "idas"
for fun, variable in zip([fun_eval_ode, fun_eval_dae], [u, g]) :
    obj = fun(False, integrator_name)["x"] - fun(True, integrator_name)["x"]
    solver = nlpsol("s", "ipopt", dict(x=variable, f=sumsqr(obj)))
    print(solver(x0=[1], lbx=[1], ubx=[3]))

print(ca.__version__)
```

**Comment by kozachynskyi** [comment]
As of branch https://github.com/casadi/casadi/compare/issue_3130 the issue persists. To be clear about issue:

IPOPT with IDAS needs more iterations in 3.6.1 than in 3.5.5. However, as you can see in the output below, IPOPT + IDAS + ODE model works as expected and IPOPT + IDAS + DAE does not. The objective function in the example depends only on state variables and is not algebraic.

I think the problem should be somewhere in the way IDAS treats ODE and DAE models differently. Unfortunately, it's hard to analyze the difference in solver outputs, as I did https://github.com/casadi/casadi/issues/3123#issuecomment-1515082840, because of the different outputs in 3.5 and 3.6.

**3.5.5 ODE**
```
iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
   0  3.5792571e+00 0.00e+00 6.26e+00  -1.0 0.00e+00    -  0.00e+00 0.00e+00   0
   1  2.6610850e+00 0.00e+00 4.73e+00  -1.0 1.55e-01    -  1.53e-01 1.00e+00f  1
   2  9.2184796e-01 0.00e+00 5.46e+00  -1.0 1.22e+00    -  1.00e+00 1.00e+00f  1
   3  7.0713233e-02 0.00e+00 1.05e+00  -1.0 2.68e-01    -  8.41e-01 1.00e+00f  1
   4  4.0419092e-04 0.00e+00 9.24e-02  -1.0 1.05e-01    -  1.00e+00 1.00e+00f  1
   5  2.7722133e-07 0.00e+00 5.30e-04  -2.5 8.58e-03    -  1.00e+00 1.00e+00f  1
   6  1.2323336e-13 0.00e+00 3.67e-07  -3.8 2.31e-04    -  1.00e+00 1.00e+00f  1
   7  4.5636321e-23 0.00e+00 1.51e-11  -5.7 1.54e-07    -  1.00e+00 1.00e+00f  1
   8  1.5912154e-30 0.00e+00 5.51e-15  -8.6 2.97e-12    -  1.00e+00 1.00e+00f  1

Number of Iterations....: 8
```

**3.6.1-9.5aa5fe218 ODE**

```
iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
   0  3.5792571e+00 0.00e+00 6.26e+00  -1.0 0.00e+00    -  0.00e+00 0.00e+00   0
   1  2.6610850e+00 0.00e+00 4.73e+00  -1.0 1.55e-01    -  1.53e-01 1.00e+00f  1
   2  9.2184873e-01 0.00e+00 5.46e+00  -1.0 1.22e+00    -  1.00e+00 1.00e+00f  1
   3  7.0713343e-02 0.00e+00 1.05e+00  -1.0 2.68e-01    -  8.41e-01 1.00e+00f  1
   4  4.0419241e-04 0.00e+00 9.24e-02  -1.0 1.05e-01    -  1.00e+00 1.00e+00f  1
   5  2.7722443e-07 0.00e+00 5.30e-04  -2.5 8.58e-03    -  1.00e+00 1.00e+00f  1
   6  1.2365657e-13 0.00e+00 3.70e-07  -3.8 2.31e-04    -  1.00e+00 1.00e+00f  1
   7  9.4374850e-23 0.00e+00 1.71e-12  -5.7 1.54e-07    -  1.00e+00 1.00e+00f  1
   8  4.2638675e-30 0.00e+00 9.01e-15  -8.6 4.26e-12    -  1.00e+00 1.00e+00f  1

Number of Iterations....: 8
```

**3.5.5 DAE**

```
iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
   0  5.1164534e-01 0.00e+00 1.04e+00  -1.0 0.00e+00    -  0.00e+00 0.00e+00   0
   1  4.0530430e-01 0.00e+00 4.22e-01  -1.0 1.08e-01    -  5.44e-01 1.00e+00f  1
   2  7.5564334e-03 0.00e+00 1.18e-02  -1.7 7.61e-01    -  1.00e+00 1.00e+00f  1
   3  3.1655261e-04 0.00e+00 2.32e-04  -1.7 9.63e-02    -  1.00e+00 1.00e+00f  1
   4  4.8758340e-07 0.00e+00 1.62e-05  -2.5 2.38e-02    -  1.00e+00 1.00e+00f  1
   5  1.4560121e-11 0.00e+00 2.55e-07  -3.8 9.68e-04    -  1.00e+00 1.00e+00f  1
   6  1.0535760e-16 0.00e+00 1.64e-08  -5.7 5.34e-06    -  1.00e+00 1.00e+00f  1
   7  4.7997981e-21 0.00e+00 1.00e-10  -8.6 1.44e-08    -  1.00e+00 1.00e+00f  1

Number of Iterations....: 7
```

**3.6.1-9.5aa5fe218 DAE**

```
iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
   0  5.1164534e-01 0.00e+00 1.04e+00  -1.0 0.00e+00    -  0.00e+00 0.00e+00   0
   1  4.0424465e-01 0.00e+00 3.92e-01  -1.0 1.09e-01    -  5.12e-01 1.00e+00f  1
   2  1.2983017e-03 0.00e+00 6.29e-01  -1.7 2.86e+00    -  1.20e-01 3.25e-01f  2
   3  7.4238097e-04 0.00e+00 5.12e-02  -1.7 8.83e-02    -  9.37e-01 1.00e+00f  1
   4  3.6648001e-04 0.00e+00 3.55e-02  -2.5 1.61e+00    -  3.97e-01 4.01e-02f  5
   5  1.4389935e-05 0.00e+00 6.17e-03  -2.5 3.35e+00    -  2.44e-01 9.55e-03f  6
   6  3.3755642e-06 0.00e+00 2.69e-03  -3.8 6.72e+00    -  1.32e-01 1.17e-03f  8
   7  9.4203617e-07 0.00e+00 1.39e-03  -3.8 5.91e+01    -  1.73e-02 6.62e-05f  9
   8  1.8657429e-07 0.00e+00 6.20e-04  -5.7 1.25e+02    -  7.99e-03 1.57e-05f 10
   9  7.2214894e-08 0.00e+00 3.86e-04  -5.7 6.43e+00  -4.0 1.00e+00 1.52e-04f 11
iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
  10  6.6469516e-09 0.00e+00 1.20e-04  -5.7 6.66e+01    -  1.50e-02 7.33e-06f 12
  11  3.6147992e-11 0.00e+00 8.68e-06  -5.7 4.54e+01    -  2.18e-02 2.69e-06f 14
  12  2.4297040e-11 0.00e+00 7.08e-06  -8.6 9.40e+01    -  1.05e-02 1.62e-07f 17
  13  1.9212998e-11 0.00e+00 6.30e-06  -8.6 2.13e-01  -4.5 1.00e+00 6.10e-05f 15
  14  1.1826252e-12 0.00e+00 1.58e-06  -8.6 1.48e+02    -  6.77e-03 5.16e-08f 18
  15  2.1109422e-13 0.00e+00 6.64e-07  -8.6 1.41e-01  -5.0 1.00e+00 1.53e-05f 17
  16  5.0350856e-14 0.00e+00 3.28e-07  -8.6 8.04e+01    -  1.24e-02 1.19e-08f 21
  17  1.3813266e-14 0.00e+00 1.70e-07  -8.6 1.93e+02    -  5.18e-03 2.47e-09f 22
  18  2.8547683e-15 0.00e+00 7.93e-08  -8.6 1.66e+02    -  1.00e+00 1.44e-09f 23
  19  1.0271963e-15 0.00e+00 4.70e-08  -8.6 3.22e+01    -  3.11e-02 3.70e-09f 24
iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls
  20  1.1427751e-16 0.00e+00 1.55e-08  -8.6 4.44e+01    -  2.38e-02 1.34e-09f 25
  21  5.0297661e-19 0.00e+00 1.02e-09  -9.0 4.17e-03  -5.4 1.00e+00 3.81e-06f 19

Number of Iterations....: 21
```

**Comment by jaeandersson** [comment]
Alright, let's do a git-bisect and find out what revision between 3.5.5 and now causes the number of iterations in IPOPT to increase.

**Comment by jgillis** [comment]
Please keep in mind that IPOPT has been updated. It's better to do such bisect with another solver.

**Comment by kozachynskyi** [comment]
> Alright, let's do a git-bisect and find out what revision between 3.5.5 and now causes the number of iterations in IPOPT to increase.

I'll try git-bisect and let you know 
@jgillis Thank you, I cannot find alternative nlpsol yet, maybe just generating jacobian and hessian will do the trick ... I'll look into it

**Comment by jaeandersson** [comment]
> Please keep in mind that IPOPT has been updated. It's better to do such bisect with another solver.

If you do git bisect with IPOPT compiled from sources, there is no need to recompile IPOPT so the version stays the same.

**Comment by jaeandersson** [comment]
OK, doing a git-bisect on
```python
from casadi import *

g = MX.sym("g")

def fun_eval_dae(fixed, integrator_name):
    L = 5.0

    #! differential states
    x = MX.sym("x")
    y = MX.sym("y")
    u = MX.sym("u")
    v = MX.sym("v")

    #! algebraic states
    lambd = MX.sym("lambda")

    #! All states and parameters
    x_all = vertcat(x, u, y, v)
    z_all = lambd

    #! We construct the DAE system
    ode = vertcat(u, lambd * x)
    alg = vertcat(
        x**2 + y**2 - L**2,
        u * x + v * y,
        u**2 - g * y + v**2 + L**2 * lambd,
    )
    x_all = vertcat(x, u)
    z_all = vertcat(y, v, lambd)
    dae = {"x": x_all, "z": z_all, "p": vertcat(g), "ode": ode, "alg": alg}

    #! the initial state of the pendulum
    X_ = [3, -1.0 / 3]
    Z_ = [4, 1.0 / 4, 1147.0 / 720]

    opts = {"t0": 0, "tf": 0.1}

    I = integrator("I", integrator_name, dae, opts)
    x0 = X_
    z0 = Z_
    all_x = []
    all_z = []

    if fixed:
        p = 2
    else:
        p = g

    for i in range(8):
        one_step = I(p=p, x0=x0, z0=z0)
        x0 = one_step["xf"]
        z0 = one_step["zf"]
        all_x.append(x0)
        all_z.append(z0)

    res = {"x": hcat(all_x), "z": hcat(all_z)}
    return res

u = MX.sym("u")


integrator_name = "idas"
for fun, variable in zip([fun_eval_dae], [g]) :
    obj = fun(False, integrator_name)["x"] - fun(True, integrator_name)["x"]
    solver = nlpsol("s", "ipopt", dict(x=variable, f=sumsqr(obj)))
    print(solver(x0=[1], lbx=[1], ubx=[3]))
```

**Comment by jaeandersson** [comment]
Git-bisect results:

af60b3cee1f9451d8f271fccd5dce6bd2c8f7ed1 is the first bad commit
commit af60b3cee1f9451d8f271fccd5dce6bd2c8f7ed1
Author: Joel Andersson <joel@jaeandersson.com>
Date:   Tue Mar 28 16:06:58 2023 -0500

    Issue #3047 Smaller backward linear system in Sundials
    No longer grows with number of adjoint sensitivities

 casadi/interfaces/sundials/cvodes_interface.cpp   |  4 +-
 casadi/interfaces/sundials/idas_interface.cpp     | 18 +++-----
 casadi/interfaces/sundials/sundials_interface.cpp | 52 +++++++++++------------
 3 files changed, 33 insertions(+), 41 deletions(-)

**Comment by jaeandersson** [comment]
Issue should be fixed by [8a3a7d5](https://github.com/casadi/casadi/commit/8a3a7d5733189f0a034d023c9559aee736cf77a4)

**Comment by kozachynskyi** [comment]
@jaeandersson Niice, it works! Thank you very much for fast support!
